shell/platform/embedder/embedder_render_target.h

```diff
@@ -77,6 +79,40 @@ class EmbedderRenderTarget {
   ///
   const FlutterBackingStore* GetBackingStore() const;
 
+  //----------------------------------------------------------------------------
+  /// @brief      Make the render target current.
+  ///
+  ///             Sometimes render targets are actually (for example)
+  ///             EGL surfaces instead of framebuffers or textures.
+  ///             In that case, we can't fully wrap them as SkSurfaces, instead,
+  ///             the embedder will provide a callback that should be called
+  ///             when the target surface should be made current.
+  ///
+  /// @return     A pair of booleans. The first bool is true if the operation
+  ///             succeeded (even if it was a no-op), false if the target could
+  ///             not be make current.
+  ///             The second boolean is true if any native graphics API
+  ///             (e.g. GL, but not EGL) state has changed and skia/impeller
+  ///             should not assume any GL state values are the same as before
+  ///             MaybeMakeCurrent was called.
+  virtual std::pair<bool, bool> MaybeMakeCurrent() const {
+    return {true, false};
```

Comment:
It's internal but I'd be tempted to create a struct with named fields rather than using a std::pair, for readability at the usage site. @chinmaygarde thoughts/preferences?

Reply:
I'd just use a struct with named fields instead.
```
struct SetCurrentResult {
  // The docs you already wrote.
  bool target_current;
  // The docs you already wrote.
  bool gl_state_trampled;
};
```

Reply:
okay, implemented that